src/types/resolveStatements.ts

```diff
@@ -201,7 +201,10 @@ function processStatements(
                 }
                 sctx = addVariable(s.name, variableType, sctx);
             } else {
-                if (expressionType.kind === "null") {
+                if (
+                    expressionType.kind === "null" ||
+                    expressionType.kind === "void"
```

Comment:
Actually we *can* infer the type `void` here, we just don't allow it in expressions other than function calls in expression statements, so the error message should reflect that